File: Deepspace_Java/P4/src/deepspace/PowerEfficientSpaceStation.java
/**
 * @author BOLÍVAR PELÁEZ, CLARA
 * @author RUIZ GÓMEZ, SOLEDAD
 * 2º DGIIM - PDOO - Practice 1 DeepSpace
 */
package deepspace;

/**
 * @title NumericDamage
 * @brief It represents the damage that an enemy starship makes to a
 * spacial station when it loses a combat
 */
public class PowerEfficientSpaceStation extends SpaceStation {
    
    private static final float EFFICIENCYFACTOR=1.10f;
    
    PowerEfficientSpaceStation (SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    Transformation setLoot(Loot loot){
        
        Transformation output = super.setLoot(loot);
        if (output == Transformation.SPACECITY)
            return Transformation.NOTRANSFORM;
        else
            return output;        
    }
    
    /**
    * @brief builds a new PowerEfficientSpaceStationToUI object from *this
    * @return WeaponToUI
    */
    PowerEfficientSpaceStationToUI getUIversion () {
        return new PowerEfficientSpaceStationToUI (this);
    }
    
    
    
    
}

File: Deepspace_Ruby/P4/PowerEfficientSpaceStation.rb
# BOLÍVAR PELÁEZ, CLARA
# RUIZ GÓMEZ, SOLEDAD
# 2º DGIIM - PDOO - Practice 4 DeepSpace

# PowerEfficientSpaceStation
# 

require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace

    class PowerEfficientSpaceStation < SpaceStation

        attr_reader :collaborators

        @@EFFICIENCYFACTOR = 1.10
        
        # Constructores
        def initialize (s)
            newCopy(s)
        end

        # fire
        def fire 
            super*@@EFFICIENCYFACTOR
        end

        # protection
        def protection 
            super*@@EFFICIENCYFACTOR
        end

        # setLoot
        def setLoot (loot)
            transformation = super
            
            if (transformation == Transformation::SPACECITY)
                return Transformation::NOTRANSFORM
            else
                return transformation
            end
        end

        # ToUI
        # builds a new PowerEfficientSpaceStationToUI object from self
        def getUIversion
            PowerEfficientSpaceStationToUI.new(self)
        end

    end
end
